### DISCUSSION TITLE
OpenSubsonic support

### DISCUSSION THREAD

**Discussion by ghost** [jeffvli/feishin#185] [General]
(I'll write this post in such a way that other people that'll perhaps ask about subsonic will get up to speed about the whole story - please correct the parts that are wrong, I'm just describing it how I see it from outside)

Hi.

# The FAQ note

I've [read the note in the FAQ](https://github.com/jeffvli/feishin#what-music-servers-does-feishin-support) about Subsonic servers not currently being supported by Feishin. This note exists because at the time of writing, the maintainers of various Subsonic servers had not decided on the way that OpenSubsonic will be implemented.

This left me intrigued, as I haven't heard about OpenSubsonic before. I've only used servers that support the original, old, Subsonic API. I was wondering what was the reason behind waiting for the new OpenSubsonic. After all, a lot more people use the original, old Subsonic API.

# Why OpenSubsonic instead of original Subsonic API
In the same Symfonium subforum that the FAQ links to, we can [find  another thread](https://support.symfonium.app/t/expanding-subsonic-api/1115) with this quote:

> Since Subsonic is dead, the official API won’t move forward and prevent clients from being able to implement advanced things that are possible with other type of Media Providers like Plex or Emby / Jellyfin.

After reading that, it now all makes sense. Feishin is a rewrite of sonixd from scratch, so it was meant to be a fresh start. The wait was because it didn't make sense to support the original, old, deprecated Subsonic API, since it'd be adding support for something that's deprecated on arrival.

# OpenSubsonic extensibility decided on

[The poll in the thread on the Symfonium subforum, that Feishin's FAQ links to](https://support.symfonium.app/t/subsonic-servers-participation/1233), now seems to be closed. The result of that poll is OpenSubsonic will be built in such a way that it'll build on top of the existing Subsonic API, by extending it with optional extensions, that don't introduce breaking changes the pre-existing Subsonic API.

To make it clear who voted in the poll about OpenSubsonic:
- [Tolriq](https://github.com/Tolriq) - maintainer of [Symfonium](https://play.google.com/store/apps/details?id=app.symfonik.music.player)
- [deluan](https://github.com/deluan) - maintainer of https://github.com/navidrome/navidrome
- [lachlan-00](https://github.com/lachlan-00) - maintainer of https://github.com/ampache/ampache
- [gravelld](https://github.com/gravelld) - maintainer of https://asti.ga/ and OneMusicAPI
- [itm (epoupon?)](https://github.com/epoupon/) - maintainer of https://github.com/epoupon/lms/
- [sentriz](https://github.com/sentriz) - maintainer of https://github.com/sentriz/gonic
- [spl0k](https://github.com/spl0k) - maintainer of https://github.com/spl0k/supysonic

# Archival of Subsonic forum and move to GitHub organization

If we look at [the pinned post in the Symfonium forum](https://support.symfonium.app/t/about-the-opensubsonic-api-extensions-category/1179), we'll read that:
> The goal of this sub forum and work is now achieved with the creation of the OpenSubsonic Organisation. [...] This forum is kept for a couple of months as history, to help understand the process and some decisions made.

So the Symfonium subforum is archived, and the main point of contact is now https://github.com/opensubsonic/open-subsonic-api

# Back to Feishin

With the OpenSubsonic API had already been decided on, and the Symfonium forum being archived, I was thinking about two action points for Feishin's README:
1. Change the link from the Symfonium forum to something else (perhaps the OpenSubsonic GitHub org)
2. Change the wording from "waiting for OpenSubsonic to be finalized" to the current status of Feishin's future in regards to OpenSubsonic (which only @jeffvli knows)

**Comment by jeffvli** [comment]
OpenSubsonic has been decided on, but unfortunately there hasn't been much progress on implementation.

I've been following the OpenSubsonic process (and am part of the OpenSubsonic org), but it's been stalled on the process of:
1. Deciding which features server owners actually want to support
2. Actually conceptualizing the features into reality

I've previously suggested some methods to the org in order to potentially speed up the decision making processes involved but was unsuccessful. If new OpenSubsonic features begin to be implemented on the server-side, I will implement it here as well.
